src/controller/scan/base_controller.go

```diff
@@ -687,23 +713,111 @@ func (bc *basicController) launchScanJob(ctx context.Context, trackID string, ar
 	params[sca.JobParameterMimes] = mimes
 	params[sca.JobParameterRobot] = robotJSON
 
-	// Launch job
-	callbackURL, err := bc.config(configCoreInternalAddr)
-	if err != nil {
-		return "", errors.Wrap(err, "launch scan job")
-	}
-	hookURL := fmt.Sprintf("%s/service/notifications/jobs/scan/%s", callbackURL, trackID)
-
-	j := &jm.JobData{
+	j := &task.Job{
 		Name: job.ImageScanJob,
-		Metadata: &jm.JobMetadata{
+		Metadata: &job.Metadata{
 			JobKind: job.KindGeneric,
 		},
 		Parameters: params,
-		StatusHook: hookURL,
 	}
 
-	return bc.jc().SubmitJob(j)
+	extraAttrs := map[string]interface{}{
+		artifactIDKey:  artifact.ID,
+		robotIDKey:     robot.ID,
+		reportUUIDsKey: reportIDs,
+	}
+
+	_, err = bc.taskMgr.Create(ctx, executionID, j, extraAttrs)
+	return err
+}
+
+// listScanTasks returns the tasks of the reports
+func (bc *basicController) listScanTasks(ctx context.Context, reportUUIDs []string) ([]*task.Task, error) {
+	if len(reportUUIDs) == 0 {
+		return nil, nil
+	}
+
+	arr := make([]string, len(reportUUIDs))
+	for i, reportUUID := range reportUUIDs {
+		arr[i] = "'" + reportUUID + "'"
+	}
+
+	// condition looks like (extra_attrs->'report_uuids')::jsonb ?| array['uuid1', 'uuid2', 'uuid3']
+	// that condition means to select the task record when any element in the report_uuids of the extra_attr is in the set ('uuid1', 'uuid2', 'uuid3')
+	condition := fmt.Sprintf("(extra_attrs->'%s')::jsonb ?| array[%s]", reportUUIDsKey, strings.Join(arr, ","))
```

Comment:
the condition is used for sql or not? Or it's just an format defination.

Reply:
This condition applies a raw filter to the querysetter of task which make the SQL looks like

```sql
select * from task where id in (select id from task where (extra_attrs->'report_uuids')::jsonb ?| array['uuid1', 'uuid2']
```

Reply:
Beego not support this way in the FilterRaw, this will be removed.